### PR TITLE
bug_fix_2159

### PR DESCRIPTION
**Pull request by SadiHassan**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp [hunk 1]
```diff
@@ -348,4 +348,16 @@ String IParserKQLFunction::getExpression(IParser::Pos & pos)
...
     return arg;
 }
+
+String IParserKQLFunction::getEscapedString(const String & input)
+{
+    String output;
+    for (auto & ch: input)
+    {
+        if (ch == '\'')
+            output += '\'';
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp:358]:
```suggestion
            output += ch;
```

**Review by ltrk2 — CHANGES_REQUESTED** [review]

**Comment by SadiHassan** [comment]
@kashwy PR is rebased and tested.

**Review by kashwy — APPROVED** [review]

**Comment by ltrk2** [comment]
@SadiHassan please don't forget about the rest of the review comments.

File: src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp [hunk 2]
```diff
@@ -348,4 +348,16 @@ String IParserKQLFunction::getExpression(IParser::Pos & pos)
 
     return arg;
 }
+
+String IParserKQLFunction::getEscapedString(const String & input)
+{
+    String output;
+    for (auto & ch: input)
```
[the remark this answers is not in the document]
Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp]: Changes are applied. Thank you!

File: src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp (hunk 1 above)
Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/IParserKQLFunction.cpp]: Changes are applied. Thank you!

**Pull request merged**